src/Microsoft.DotNet.Build.Tasks.Feed/BlobFeedAction.cs

```diff
@@ -22,84 +22,64 @@ sealed class BlobFeedAction
-            }
-            else
-            { 
-                throw new Exception("Unable to parse expected feed. Please check ExpectedFeedUrl.");
-            }
+            this.feed = new BlobFeed(accountName, accountKey, containerName, packagesPath, Log);
         }
 
-        public async Task<bool> PushToFeed(IEnumerable<string> items, bool allowOverwrite = false)
+        public async Task<bool> PushToFeed(bool allowOverwrite = false)
         {
-            if (feed.IsSanityChecked(items))
+            if (feed.IsSanityChecked(feed.PackagesPath))
             {
                 if (CancellationToken.IsCancellationRequested)
                 {
                     Log.LogError("Task PushToFeed cancelled");
                     CancellationToken.ThrowIfCancellationRequested();
                 }
 
-                using (var clientThrottle = new SemaphoreSlim(this.MaxClients, this.MaxClients))
-                {
-                    await Task.WhenAll(items.Select(item => PushItemToFeed(item, feed.RelativePath, clientThrottle, allowOverwrite, false)));
-                }
+                await PushItemsToFeedAsync(feed.PackagesPath, allowOverwrite);
             }
 
             return !Log.HasLoggedErrors;
         }
 
-        public async Task<bool> PushToFeedFlat(IEnumerable<string> items, bool allowOverwrite = false)
+        public async Task<bool> PushItemsToFeedAsync(string packagesPath, bool allowOverwrite)
         {
-            if (CancellationToken.IsCancellationRequested)
-            {
-                Log.LogError("Task PushToFeedFlat cancelled");
-                CancellationToken.ThrowIfCancellationRequested();
-            }
+            Log.LogMessage(MessageImportance.Low, $"START pushing items to feed");
 
-            using (var clientThrottle = new SemaphoreSlim(this.MaxClients, this.MaxClients))
-            {
-                await Task.WhenAll(items.Select(item => PushItemToFeed(item, feed.RelativePath, clientThrottle, allowOverwrite, true)));
-            }
-            return !Log.HasLoggedErrors;
-        }
-
-        public async Task<bool> PushItemToFeed(string item, string relativePath, SemaphoreSlim clientThrottle, bool allowOverwrite, bool isFlat)
-        {
             try
             {
-                string uploadPath = feed.CalculateBlobPath(item, relativePath);
-                string packageDirectory = feed.CalculateRelativeUploadPath(item, relativePath);
-
-                await UploadAsync(CancellationToken, item, uploadPath, clientThrottle, allowOverwrite);
-
-                if (!isFlat)
+                Source source = new Source
+                {
+                    Name = feed.ContainerName,
+                    Type = "azure",
+                    Path = $"https://{feed.AccountName}.blob.core.windows.net/{feed.ContainerName}/",
```

Comment:
Why can't this just be the feed URL that we pass in instead of constructing it manually here?